src/core/createMetrics.js

```diff
@@ -371,11 +371,12 @@ export class Metrics extends EventEmitter {
      * @returns {Promise}
      * @private
      */
-    _addEventNameToPromise(eventName, promise, shouldMerge) {
+    _addEventNameToPromise(eventName, promise, shouldMerge, element) {
```

Comment:
Non blocking question... would it make sense to change the `_addEventNameToPromise` signature to use a single object argument instead of 4 arguments for legibility? 

```
_addEventNameToPromise({eventName, promise, shouldMerge, element}) {
```